src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -5,13 +5,14 @@
 using System.Threading.Tasks;
 using Orleans.Concurrency;
 using Orleans.Runtime;
+using Orleans.Serialization;
 
 namespace Orleans.Streams
 {
     internal interface IStreamSubscriptionHandle
     {
         Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken);
-        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken);
+        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken, SerializationManager serializationManager);
```

Comment:
It's a little annoying that this and `IBatchContainer` take a dependency on SerializationManager, where *I believe* for most implementors this is unused, but it's still a dependency that flows through every stream provider, factory, adaptors, caches, etc (although to be fair, I scanned them pretty fast, so I'm not sure if there's a better alternative).
I suppose that as we use DI more and more, we can probably get rid of these dependencies being injected in methods, and only add them the the relevant implementations that do need them. Nevertheless, even if temporarily, it strikes me as this is a big change that touches a lot of files, but we are already talking that the streaming infrastructure shouldn't have used SerializationManager, and that we should start endorsing a more version tolerant serializer (as this **is** persisted, even if for little time in some cases). Anyway, just pointing out, in case you can think of a better alternative